From 5da7cff5cfffe74e86631981d48aa9775e087b29 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Mon, 1 Sep 2014 13:33:19 -0400
Subject: [crypto] qat: Fixed SKU1 dev issue

Message-id: <1409578401-26943-28-git-send-email-nikolay@redhat.com>
Patchwork-id: 90104
O-Subject: [RHEL7.1 PATCH BZ 1091130 27/29] crypto: qat - Fixed SKU1 dev issue
Bugzilla: 1091130
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>

From: Tadeusz Struk <tadeusz.struk@intel.com>

Fix for issue with SKU1 device.
SKU1 device has 8 micro engines as opposed to 12 in other SKUs
so it was not possible to start the non-existing micro engines.

Signed-off-by: Bo Cui <bo.cui@intel.com>
Signed-off-by: Tadeusz Struk <tadeusz.struk@intel.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit 4f74c3989b14338544b65360ca4f8587c63d7fd9)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/crypto/qat/qat_common/qat_uclo.c b/drivers/crypto/qat/qat_common/qat_uclo.c
index 2580091..1e27f9f 100644
--- a/drivers/crypto/qat/qat_common/qat_uclo.c
+++ b/drivers/crypto/qat/qat_common/qat_uclo.c
@@ -256,10 +256,6 @@ static int qat_uclo_fetch_initmem_ae(struct icp_qat_fw_loader_handle *handle,
   pr_err("QAT: Parse num for AE number failed\n");
   return -EINVAL;
  }
- if (!test_bit(*ae, (unsigned long *)&handle->hal_handle->ae_mask)) {
-  pr_err("QAT: ae %d to be init is fused off\n", *ae);
-  return -EINVAL;
- }
  if (*ae >= ICP_QAT_UCLO_MAX_AE) {
   pr_err("QAT: ae %d out of range\n", *ae);
   return -EINVAL;
@@ -456,7 +452,7 @@ static int qat_uclo_init_memory(struct icp_qat_fw_loader_handle *handle)
    (sizeof(struct icp_qat_uof_memvar_attr) *
    initmem->val_attr_num));
  }
- for (ae = 0; ae < ICP_QAT_UCLO_MAX_AE; ae++) {
+ for (ae = 0; ae < handle->hal_handle->ae_max_num; ae++) {
   if (qat_hal_batch_wr_lm(handle, ae,
      obj_handle->lm_init_tab[ae])) {
    pr_err("QAT: fail to batch init lmem for AE %d\n", ae);
diff --git a/drivers/crypto/qat/qat_dh895xcc/adf_dh895xcc_hw_data.h b/drivers/crypto/qat/qat_dh895xcc/adf_dh895xcc_hw_data.h
index c5ce236..b707f29 100644
--- a/drivers/crypto/qat/qat_dh895xcc/adf_dh895xcc_hw_data.h
+++ b/drivers/crypto/qat/qat_dh895xcc/adf_dh895xcc_hw_data.h
@@ -59,7 +59,7 @@
 #define ADF_DH895XCC_FUSECTL_SKU_4 0x3
 #define ADF_DH895XCC_MAX_ACCELERATORS 6
 #define ADF_DH895XCC_MAX_ACCELENGINES 12
-#define ADF_DH895XCC_ACCELERATORS_REG_OFFSET 18
+#define ADF_DH895XCC_ACCELERATORS_REG_OFFSET 13
 #define ADF_DH895XCC_ACCELERATORS_MASK 0x3F
 #define ADF_DH895XCC_ACCELENGINES_MASK 0xFFF
 #define ADF_DH895XCC_LEGFUSE_OFFSET 0x4C
-- 
1.7.1